runtime/sema/check_invocation_expression.go

```diff
@@ -174,6 +174,7 @@ func (checker *Checker) checkInvocationExpression(invocationExpression *ast.Invo
 	if returnType == NeverType {
 		returnInfo := checker.functionActivations.Current().ReturnInfo
 		returnInfo.DefinitelyHalted = true
```

Comment:
doesn't this directly contradict the definition of `NeverType`?

Reply:
Instead of marking this as `returnInfo.DefinitelyReturned`, can the usages be updated to use `returnInfo.DefinitelyReturned || returnInfo.DefinitelyHalted` instead of `returnInfo.DefinitelyReturned`?

Reply:
@dreamsmasher 
> doesn't this directly contradict the definition of `NeverType`?

Can you elaborate? Not quite sure I follow

Reply:
the wording is a bit confusing here, since ⊥ implies that it either doesn't halt or it panics. at the same time, the above comment mentions `Update the return info for invocations that do not return (i.e. have a `Never` return type)`. I agree with Daniel, we should either change the naming for these attributes or add comments explaining why we mark them as Definitely{Halted,Returned} here.

Reply:
Ah, I see how it's confusing, thank you for the explanation. I'll try tracking a combination of halted/returned, instead of treating one as resulting in the other.